rules/aip0203/optional.go

```diff
@@ -0,0 +1,108 @@
+package aip0203
+
+import (
+	"regexp"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+	"google.golang.org/genproto/googleapis/api/annotations"
+)
+
+// This rule inspects the leading comments of each field
+// and if anything looks similar to "Optional.", it throws
+// a problem.
+//
+// Examples:
+// Incorrect code for this rule:
+//
+//	message Book {
+//	// The title of the book.
+//	// @Optional
+//	string title = 1;
+//	}
+//
+// or
+//
+//	message Book {
+//	// Optional. The title of the book.
+//	string title = 1;
+//	}
+//
+//
+// Correct code for this rule:
+//
+//	message Book {
+//		// The title of the book.
+//		string title = 1 [(google.api.field_behavior) = OPTIONAL];
+//	}
+var optional = &lint.FieldRule{
+	Name:   lint.NewRuleName("core", "0203", "optional"),
+	URI:    "http://aip.dev/203#optional",
+	OnlyIf: withoutOptionalFieldBehavior,
+	LintField: func(f *desc.FieldDescriptor) []lint.Problem {
+		return checkLeadingComments(f, optionalRegexp, "OPTIONAL")
+	},
+}
+
+var optionalBehaviorConflict = &lint.FieldRule{
+	Name: lint.NewRuleName("core", "0203", "optional-conflict"),
+	URI:  "http://aip.dev/203#optional",
+	OnlyIf: func(f *desc.FieldDescriptor) bool {
+		return !withoutOptionalFieldBehavior(f)
+	},
+	LintField: func(f *desc.FieldDescriptor) []lint.Problem {
+		// APIs may use the OPTIONAL value to describe a field which doesn't use
+		// REQUIRED, IMMUTABLE, INPUT_ONLY or OUTPUT_ONLY. If a field is described
+		// as optional, it can't be others.
+		if len(utils.GetFieldBehavior(f)) > 1 {
+			return []lint.Problem{{
+				Message:    "Field behavior `(google.api.field_behavior) = OPTIONAL` shouldn't be used together with other field behaviors.",
+				Descriptor: f,
+			}}
+		}
+		return nil
+	},
+}
+
+// If a field is described as optional, ensure that every optional field on the
```

Comment:
Is it a typo in `If a field`? Should it start with `If a message`?

Reply:
To make it more clear, change it "If a message has a field which is described as optional"